test/role.test.js

```diff
@@ -489,48 +489,230 @@ describe('role model', function() {
     });
   });
 
-  it('resolves OWNER via "userId" property with no relation', function() {
-    var Album = app.registry.createModel('Album', {
-      name: String,
-      userId: Number,
-    });
-    app.model(Album, {dataSource: 'db'});
+  it('should support legacy owner role resolver', function() {
+    var password = 'pass';
+    var user, sender, userId, senderId;
 
-    var user;
-    return User.create({email: 'test@example.com', password: 'pass'})
-      .then(u => {
-        user = u;
-        return Album.create({name: 'Album 1', userId: user.id});
-      })
-      .then(album => {
-        return Role.isInRole(Role.OWNER, {
-          principalType: ACL.USER, principalId: user.id,
-          model: Album, id: album.id,
-        });
-      })
-      .then(isInRole => expect(isInRole).to.be.true());
+    var Message = app.registry.createModel('message', {
+      content: String,
+    }, {
+      relations: {
+        user: {
+          type: 'belongsTo',
+          model: 'User',
+          foreignKey: 'userId',
+        },
+        sender: {
+          type: 'belongsTo',
+          model: 'User',
+          foreignKey: 'senderId',
+        },
+      },
+    });
+    app.model(Message, {dataSource: 'db'});
+
+    return app.models.User.create([
+        {email: 'user@user.com', password},
+        {email: 'sender@sender.com', password},
+    ])
+    .then(users => {
+      user = users[0];
+      sender = users[1];
+      userId = user.id;
+      senderId = sender.id;
+
+      var messages = [
+        {userId},
+        {senderId},
+        {},
+      ];
+      return Promise.map(messages, msg => {
+        return Message.create(msg);
+      });
+    })
+    .then(messages => {
+      var userAccessContext = {
+        principalType: ACL.USER,
+        principalId: user.id,
+        model: Message,
+        id: messages[0].id,
+      };
+      var senderAccesContext = {
+        principalType: ACL.USER,
+        principalId: sender.id,
+        model: Message,
+        id: messages[1].id,
+      };
+      var noOwnerAccessContext = {
+        principalType: ACL.USER,
+        principalId: sender.id,
+        model: Message,
+        id: messages[2].id,
+      };
+
+      return Promise.all([
+        Role.isInRole(Role.OWNER, userAccessContext),
+        Role.isInRole(Role.OWNER, senderAccesContext),
+        Role.isInRole(Role.OWNER, noOwnerAccessContext),
+      ]);
+    })
+    .spread((isUserOwner, isSenderOwner, isSenderAlsoOwner) => {
+      expect(isUserOwner).to.be.true();
+      expect(isSenderOwner).to.be.false();
+      expect(isSenderAlsoOwner).to.be.false();
```

Comment:
One assert per test please. Ideally, the original "resolves OWNER via ..." tests should be preserved.

If you are concerned about code duplication, then please extract shared helpers, or perhaps move these tests inside a `describe` block where the shared setup can be performed via `beforeEach` hook.